webapp/src/components/backstage/automation/patterned_text_area.tsx

```diff
@@ -33,7 +33,7 @@ export const PatternedTextArea = (props: Props) => {
             return;
         }
 
-        if (props.maxRows && urls.length > props.maxRows) {
+        if (props.maxRows && urls.split(props.delimiter || '\n').filter((v) => v.trim().length).length > props.maxRows) {
```

Comment:
Super nit: is it possible to return a literal boolean in the filter function? Something like this:
```suggestion
        if (props.maxRows && urls.split(props.delimiter || '\n').filter((v) => v.trim().length > 0).length > props.maxRows) {
```

I sometimes find difficult to read when these implicit conversions to boolean happen.